sway/tree/output.c

```diff
@@ -60,6 +60,21 @@ static void restore_workspaces(struct sway_output *output) {
 		struct sway_workspace *ws = root->noop_output->workspaces->items[0];
 		workspace_detach(ws);
 		output_add_workspace(output, ws);
+		
+		// If the floater was made floating while on the NOOP output, its width
+		// and height will be zero and it should be reinitialized as a floating
+		// container to get the appropriate size and location. Additionally, if
+		// the floater's top-left corner is outside of the workspace, do the
+		// same as the output layout has likely changed and the maximum size
+		// may be different
+		for (int i = 0; i < ws->floating->length; i++) {
+			struct sway_container *floater = ws->floating->items[i];
+			if (floater->width == 0 || floater->height == 0 ||
+					floater->x < ws->x || floater->y < ws->y) {
```

Comment:
I don't think this check is right/enough.

Shouldn't we also check for the right/bottom borders? Shouldn't users be able to drag the top-left corner past the workspace bounds?

Reply:
I think it should be good now. I changed it to use the output bounds instead of the workspace bounds and added a check for the bottom/right borders

Reply:
One question remains: do we want to allow this configuration?

```
floating
+-------------------------+
|                         |
|                         |
|                         |
|         +------------------------------------------+
|         |               |                          |
|         |               |                          |
|         |               |                          |
|         |               |                          |
+-------------------------+                          |
          |                                          |
          |                                          |
          |                                          |
          |                                          |
          |                                          |
          |                                          |
          |                                          |
          |                                          |
          |                                          |
          |                                          |
          |                                          |
          +------------------------------------------+
          workspace
```

Reply:
I changed it to check for a width/height of zero, a width/height larger than the output, or being completely outside of the output. That should allow for the above to be retained.